### ISSUE TITLE
Program doesn't switch between constant frequency and wavelength

### ISSUE THREAD

**Issue by bruceweaver** [kenwdelong/frog#2]
The default input is constant wavelength. When we try to switch to constant frequency the program still appears to read in constant wavelengths. Is there a way to fix this?

Thanks

**Comment by kenwdelong** [comment]
Why do you say it's not reading in in constant frequency?  What's the symptom?

You can see the code here: https://github.com/kenwdelong/frog/blob/master/FROG/RawData.pas#L229

If you take a very short pulse, say 10fs, generate the FROG trace, and then read it back in both ways, you should see that the trace has different shapes depending on if you read it in with constant frequency or wavelength.  For pulses much longer than, say, 10-20fs, you won't be able to see the difference.

**Comment by bruceweaver** [comment]
Thanks for responding!

The UI doesn't change in any way when we switch between the two, but we input the values anyway, put 750 as the central frequency (unsure what units it is reading in when in constant frequency mode). When we retrieved the pulse, the spectrum was centred at 1500 nm, indicating that it read it as 750nm (SHG FROG so that wavelength would be doubled.

**Comment by kenwdelong** [comment]
That's correct, neither the UI nor the central wavelength will be affected by that setting.

What that setting is doing is changing how to interpret how the points are to be interpreted along the wavelength/frequency axis of the FROG trace being read in.

Let's assume the central wavelength of the trace is 800nm (meaning for SHG the fundamental was 1600nm).

On constant frequency (assuming delta_lam = 1nm), the points would be interpreted as (starting in the center of the grid): 800nm, 801nm, 802nm, 803nm (or in frequency, c/800, c/801, c/802, c/803).  Because the points are not on a constant frequency spacing, but the FFT requires constant frequency, they will be interpolated nonlinearly onto a constant frequency grid.

If the setting is on constant frequency, with the central frequency is c/800 and the spacing as del_f, the points are understood as c/800, c/800 + del_f, c/800 + 2*del_f, c/800 + 3*del_f, etc.  So they don't need nonlinear interpolation when read in.

The only way this makes much difference is with very short pulses (very broad spectrum).  You can use real units, create a 10fs pulse, and save the FROG trace.  Then try reading it in with constant frequency and constant wavelength.  At that bandwidth, you should see a difference.  Actually, if you use SHG, you might not be able to see it easily (since everying is an oval).  Try PG FROG.  Also, retrieve the pulse and see what the results are.
